packages/jest-snapshot/src/index.js

```diff
@@ -22,21 +22,34 @@ import * as utils from './utils';
 const fileExists = (filePath: Path, hasteFS: HasteFS): boolean =>
   hasteFS.exists(filePath) || fs.existsSync(filePath);
 
-const cleanup = (hasteFS: HasteFS, update: SnapshotUpdateState) => {
+const cleanup = (
+  hasteFS: HasteFS,
+  update: SnapshotUpdateState,
+  snapshotTag: ?string,
+) => {
   const pattern = '\\.' + utils.SNAPSHOT_EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
   const filesRemoved = files
-    .filter(
-      snapshotFile =>
-        !fileExists(
-          path.resolve(
-            path.dirname(snapshotFile),
-            '..',
-            path.basename(snapshotFile, '.' + utils.SNAPSHOT_EXTENSION),
-          ),
-          hasteFS,
+    .filter(snapshotFile => {
+      if (snapshotTag) {
+        const baseNameParts = path.basename(snapshotFile).split('.');
+        baseNameParts.splice(snapshotTag.split().length * -1 - 1);
```

Comment:
ugh, I don't like `splice` at all, can we not mutate this array and use yet another variable?
Also, I feel like there's a nice opportunity to extract a function for finding snapshot associated with a file (some of the logic in line 37 and 44 is quite similar).

Reply:
Hey @thymikee! well, first of all, I'm all open to suggestions. The idea on the splice is the following:
```
I want to run my tests in two projects. One for iOS, one for Android. So my snapshotTags will be "ios" and "android".
What happens in this function is that it removes the "snapshot extension" to try to find out if there's a test file related to this snapshot or not. Now, let's imagine this:

./__tests__/RoundButton.js
./__tests__/__snapshots__/RoundButton.js.ios.snap -> created when running the project with ios tag
./__tests__/__snapshots__/RoundButton.js.android.snap -> created when running the project with android tag
./__tests__/__snapshots__/RoundButton.js.snap -> older test, when I didn't have tags yet.

for each time I run the tests, I know only the current snaptshotTag, so if I have a snapshotTag, I need to extract it from the name the snapshot (which in this case I tried to allow to have dots) and the extension. As I have no idea about how many dots will have, in the path and in the snapshotTag, I used splice and split to remove the needed elements...
```

But as I write, I'm thinking in a different approach using only split. Let me try something...